hls4ml/model/optimizer/__init__.py

```diff
@@ -33,6 +33,7 @@
 register_flow(
     'convert',
     [
+        'infer_precision_types',
```

Comment:
I would prefer having this towards the end of convert, or just leaving the call out of convert, since we call it in the optimize flow anyway. What is the purpose of doing it here? Here it messes up some of the conversion steps I have for qonnx, which only happen if the type is not set, since it doesn't want to override set types, and this inferring here sets the types.

Reply:
"optimize" is sort-of optional, it belongs to "convert" because after that stage the other optimizers don't expect "auto" to exist. If onnx has its own optimizers that run, why not group them into a flow and run before convert? the idea is that after "convert" it shouldn't matter where the model came from

Reply:
But we can move it later in concert, right?

Reply:
If you have onnx-specific optimizers that must go before, place them before this one.